### PR TITLE
Minor refactor to research, removing some magic numbers and a sub_<hex>

### PR DESCRIPTION
**Pull request by rwjuk**

### CODE REVIEW

**Comment by Gymnasiast** [comment]
Comment from the sub naming PR: https://github.com/OpenRCT2/OpenRCT2/pull/5532#discussion_r119876199:

The following should be put into a separate function (if one does not exist):
```
for (sint32 i = 0; i < 4; i++) {
			gResearchedRideTypes[i] = 0xFFFFFFFF;
		}

		for (sint32 i = 0; i < 8; i++) {
			gResearchedRideEntries[i] = 0xFFFFFFFF;
		}
```

**Comment by rwjuk** [comment]
This can go in as-is if you want, I can put the other refactoring in a new PR. I don't mind either way.

**Comment by rwjuk** [comment]
Done!

File: src/openrct2/management/research.c [hunk 1]
```diff
@@ -643,3 +643,14 @@ void reset_researched_scenery_items()
 		gResearchedSceneryItems[i] = 0xFFFFFFFF;
 	}
 }
+
+void reset_researched_ride_types_and_entries()
+{
+	for (sint32 i = 0; i < 4; i++) {
```
Comment (rwjuk) [inline src/openrct2/management/research.c]: Not sure why this uses 4, when the maximum number of researched ride types is actually 8....

Comment (Gymnasiast) [inline src/openrct2/management/research.c]: Change it, but leave a note saying
```
Used to be 4 for unknown reasons, likely a mistake
```
or something similar.

Comment (rwjuk) [inline src/openrct2/management/research.c]: Done.

**Pull request merged**